src/workflow/StreamlitUI.py

```diff
@@ -28,12 +37,32 @@ def __init__(self, workflow_dir, logger, executor, paramter_manager):
         self.parameter_manager = paramter_manager
         self.params = self.parameter_manager.get_parameters_from_json()
 
+    def tk_directory_dialog(self, title: str = "Select Directory", parent_dir: str = os.getcwd()):
+        """
+        Creates a Tkinter directory dialog for selecting a directory.
+
+        Args:
+            title (str): The title of the directory dialog.
+            parent_dir (str): The path to the parent directory of the directory dialog.
+
+        Returns:
+            str: The path to the selected directory.
+        
+        Warning:
+            This function is not avaliable in a streamlit cloud context.
+        """
+        root = Tk()
```

Comment:
```suggestion
        root = Tk()
        root.attributes("-topmost", True)
```

Tkinter is run from the console and not the browser window, which is why it can launch behind the browser. This attribute ensures that the file dialog is displayed above all other windows.